cirq-core/cirq/ops/measurement_gate.py

```diff
@@ -126,9 +126,12 @@ def full_invert_mask(self):
     def _is_measurement_(self) -> bool:
         return True
 
-    def _measurement_key_name_(self):
+    def _measurement_key_name_(self) -> str:
         return self.key
 
+    def _measurement_key_obj_(self) -> value.MeasurementKey:
+        return self.mkey
```

Comment:
This `self.(key|mkey)` differs from `self._key` used in the (more recent) `kraus_channel.py` file. Do we plan to converge on one format or the other?

Reply:
I think `key` and `mkey` are better names and more indicative of the type. But updating the channels would also entail inward-ing and recreating the JSON serializations. I don't know if the renaming is worth that. Also, the public interfaces of these objects are different. The channels don't expose a `key` attribute but `MeasurementGate` has to continue exposing it *as a string* or do a deprecation cycle on it.  WDYT?

Reply:
Sounds like there's some complexity here - no need to handle it in this PR. Could you open an issue to track it?